src/pydocstyle/parser.py

```diff
@@ -117,7 +119,36 @@ def is_public(self):
 
         This helps determine if it requires a docstring.
         """
-        return not self.name.startswith('_') or self.name.startswith('__')
+        module_name = Path(self.name).stem
+        return (
+            not self._is_inside_private_package() and
+            self._is_public_name(module_name)
+        )
+
+    def _is_inside_private_package(self):
+        """Return True if the module is inside a private package."""
+        path = Path(self.name).parent  # Ignore the actual module's name
+        syspath = [Path(p) for p in sys.path]  # Convert to pathlib.Path.
+
+        # Bail if we are at the root directory or in `PYTHONPATH`.
+        while path != path.parent and path not in syspath:
```

Comment:
I don't think we should iterate all the way to the root. But only till the current directory? Thoughts?

Reply:
edit - nvm, the current working directory should always be in the sys path so we should be fine.

Reply:
I did some tests with the `pydocstyle` CLI locally, and the directory that you execute `pydocstyle` from is **not included** in `sys.path`.

> I don't think we should iterate all the way to the root. But only till the current directory? 

Maybe your right? Please confirm and advise.

Reply:
I am just wondering about this use case because if we iterate all they way till root, there might be folders with an underscore in their name. On the other hand this means that pydocstyle will have different outputs on the same input based on the `CWD` which I am not sure about. So - let's keep it to `sys.path` for now simply because it makes the checker consistent.

In the future if we decide to change this behavior we can always put in a PR - @Nurdok what do you think?